internal/common/tree.go

```diff
@@ -176,84 +239,63 @@ func RemoveLabelFromTree(node *DNSTreeNode, label string) {
 }
 
 func AddLabelToTree(node *DNSTreeNode, label, value string) {
+	if len(node.Endpoints) == 0 {
+		node.Endpoints = append(node.Endpoints, &endpoint.Endpoint{Labels: map[string]string{label: value}})
+	} else {
+		for _, ep := range node.Endpoints {
+			if ep.Labels == nil {
+				ep.Labels = endpoint.NewLabels()
+			}
+			ep.Labels[label] = value
+		}
+	}
 	for _, c := range node.Children {
 		AddLabelToBranch(node, c.Name, label, value)
 		AddLabelToTree(c, label, value)
 	}
 }
 
-func HasLabelForBranch(node *DNSTreeNode, branch, label, value string) bool {
-	d := findDataSetForChild(node, branch)
-	if d == nil {
-		return false
-	}
-	if v, ok := d.Labels[label]; ok {
-		return value == v
+func hasLabelForBranch(node *DNSTreeNode, branch, label, value string) bool {
+	for _, ep := range findEndpointsForChild(node, branch) {
+		if v, ok := ep.Labels[label]; ok {
+			if value == v {
```

Comment:
is 
```
if value == v {
	return true
} else {
	return false
}
```

equivalent to `return value == v `?

Reply:
nice, cheers. I assume I had it like that while I was wanting to log the decisions, but I'll change it now :)